subsys/mgmt/mcumgr/smp_bt.c

```diff
@@ -41,6 +56,101 @@ static struct bt_uuid_128 smp_bt_svc_uuid = BT_UUID_INIT_128(
 static struct bt_uuid_128 smp_bt_chr_uuid = BT_UUID_INIT_128(
 	BT_UUID_128_ENCODE(0xda2e7828, 0xfbce, 0x4e01, 0xae9e, 0x261174997c48));
 
+/* Helper function that allocates conn_param_data for a conn. */
+static struct conn_param_data *alloc_conn_param_data(struct bt_conn *conn)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(conn_data); i++) {
+		if (conn_data[i].conn == NULL) {
+			conn_data[i].conn = conn;
+			return &conn_data[i];
+		}
+	}
+
+	/* Conn data must exists. */
+	__ASSERT_NO_MSG(false);
+	return NULL;
+}
+
+/* Helper function that returns conn_param_data associated with a conn. */
+static struct conn_param_data *get_conn_param_data(const struct bt_conn *conn)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(conn_data); i++) {
+		if (conn_data[i].conn == conn) {
+			return &conn_data[i];
+		}
+	}
+
+	/* Conn data must exists. */
+	__ASSERT_NO_MSG(false);
+	return NULL;
+}
+
+/* Work handler function for restoring the default latency for the connection. */
+static void restore_default_latency(struct k_work *work)
+{
+	struct bt_le_conn_param params;
+	struct conn_param_data *cpd;
+	struct bt_conn_info info;
+	int ret = 0;
+
+	cpd = CONTAINER_OF(work, struct conn_param_data, dwork);
+
+	if (cpd->latency_state & CONN_LOW_LATENCY_REQUESTED) {
+		(void)k_work_reschedule(&cpd->dwork, K_MSEC(RESTORE_TIME));
+		cpd->latency_state &= ~CONN_LOW_LATENCY_REQUESTED;
+	} else {
+		ret = bt_conn_get_info(cpd->conn, &info);
+		if (ret != 0) {
```

Comment:
Then we would be stuck with low latency ...

Reply:
I think we may either explicitly assume this never happens (I think it would actually never happen looking at the implementation) or reschedule work here

Reply:
We could assert or ignore the returned value if we know that given code path would never be taken